src/arrayMethodSort.js

```diff
@@ -4,9 +4,35 @@
  * Implement method Sort
  */
 function applyCustomSort() {
+
   [].__proto__.sort2 = function(compareFunction) {
     // write code here
+    for (let i = 0; i < this.length; i++) {
+      if (compareFunction) {
+        const comparisonRes = compareFunction(this[i], this[i + 1]);
+        if (comparisonRes > 0) {
+          const temp = this[i];
+          this[i] = this[i + 1];
+          this[i + 1] = temp;
+        }
+      }
+      else {
```

Comment:
The else statement does not have a corresponding if statement. Also, the logic in the else block is incorrect for sorting when no compareFunction is provided. It should sort based on string conversions of the elements, not just using localeCompare.